aten/src/ATen/native/TensorCompare.cpp

```diff
@@ -23,6 +23,177 @@ void where_cpu(
+        [other_val](scalar_out& ret_val, const scalar& self_val) {
+          ret_val = at::convert<scalar_out>(std::less_equal<scalar>()(self_val, other_val));
+      }
+    );
+  }
+};
+
+template<typename scalar_out, typename scalar>
+struct CmpOp<std::greater_equal, scalar_out, scalar> {
+  static void apply(at::Tensor& ret, const at::Tensor& self, at::Scalar other) {
+    auto other_val = other.to<scalar>();
+
+    if (isIntegralType(self.type().scalarType()) && other.isFloatingPoint()) {
+      auto other_double = other.to<double>();
+      auto other_long = other.to<int64_t>();
+      if (other_double != other_long) {
+        other_val = at::convert<scalar>(ceil(other_double));
+      }
+    }
+
+    at::CPU_tensor_apply2<scalar_out, scalar>(ret, self,
+        [other_val](scalar_out& ret_val, const scalar& self_val) {
+          ret_val = at::convert<scalar_out>(std::greater_equal<scalar>()(self_val, other_val));
+      }
+    );
+  }
+};
+
+template<typename scalar_out, typename scalar>
+struct CmpOp<std::equal_to, scalar_out, scalar> {
+  static void apply(at::Tensor& ret, const at::Tensor& self, at::Scalar other) {
+    auto other_val = other.to<scalar>();
+
+    if (isIntegralType(self.type().scalarType()) && other.isFloatingPoint()) {
+      auto other_double = other.to<double>();
+      auto other_long = other.to<int64_t>();
+      if (other_double != other_long) {
+        ret.fill_(0);
+      }
+    } else {
+      at::CPU_tensor_apply2<scalar_out, scalar>(ret, self,
+          [other_val](scalar_out& ret_val, const scalar& self_val) {
+            ret_val = at::convert<scalar_out>(std::equal_to<scalar>()(self_val, other_val));
+        }
+      );
+    }
+  }
+};
+
+template<typename scalar_out, typename scalar>
+struct CmpOp<std::not_equal_to, scalar_out, scalar> {
+  static void apply(at::Tensor& ret, const at::Tensor& self, at::Scalar other) {
+    auto other_val = other.to<scalar>();
+
+    if (isIntegralType(self.type().scalarType()) && other.isFloatingPoint()) {
+      auto other_double = other.to<double>();
+      auto other_long = other.to<int64_t>();
+      if (other_double != other_long) {
+        ret.fill_(1);
+      }
+    } else {
+      at::CPU_tensor_apply2<scalar_out, scalar>(ret, self,
+          [other_val](scalar_out& ret_val, const scalar& self_val) {
+            ret_val = at::convert<scalar_out>(std::not_equal_to<scalar>()(self_val, other_val));
+        }
+      );
+    }
+  }
+};
+
+template<template<typename T> class Comparator>
+at::Tensor& cmp_out_cpu(at::Tensor& result, const at::Tensor& self, at::Scalar other, const char* op_name) {
+  result.resize_(self.sizes());
+  AT_DISPATCH_ALL_TYPES(self.type(), op_name, [&]() {
+    CmpOp<Comparator, uint8_t, scalar_t>::apply(result, self, other);
+  });
+  return result;
+}
+
+template<template<typename T> class Comparator>
+at::Tensor& cmp_out_cpu(at::Tensor& result, const at::Tensor& self, const at::Tensor& other, const char* op_name) {
```

Comment:
I think this could use the `s_` mechanism, but I'm not 100% sure how to use it with native functions.